jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -262,6 +265,29 @@ static PluginConfigurationProcessor processCommonConfiguration(
         isTargetImageCredentialPresent);
   }
 
+  /**
+   * Parses the list of raw volumes directories to a list of {@link AbsoluteUnixPath}
+   *
+   * @param rawConfiguration raw configuration data
+   * @return the list of parsed volumes.
+   * @throws InvalidContainerVolumesException if {@code volumes} are not valid absolute Unix paths
+   */
+  @VisibleForTesting
+  static List<AbsoluteUnixPath> getVolumesList(RawConfiguration rawConfiguration)
+      throws InvalidContainerVolumesException {
+    List<AbsoluteUnixPath> volumes = new ArrayList<>();
+    for (String directory : rawConfiguration.getVolumes()) {
+      try {
+        AbsoluteUnixPath absoluteUnixPath = AbsoluteUnixPath.get(directory);
+        volumes.add(absoluteUnixPath);
+      } catch (IllegalArgumentException exception) {
+        throw new InvalidContainerVolumesException("Invalid volume path: " + directory);
```

Comment:
Let's put this message in the constructor for `InvalidContainerVolumeException` (also rename to singular?). Also, the message should probably also include some indication of what a valid volume path is (ie. "(must be an absolute Unix-style path)").

Reply:
Should also include the cause (the `IllegalArgumentException`).

Reply:
Changed the name, message and added the cause to this exception.